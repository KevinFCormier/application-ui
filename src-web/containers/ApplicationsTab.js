/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import ResourceTableModule from '../components/common/ResourceTableModuleFromProps'
import { withRouter } from 'react-router-dom'
import { RESOURCE_TYPES, ROLES } from '../../lib/shared/constants'
import { typedResourcePageWithListAndDetails } from '../components/common/ResourcePage'
import { createApplication } from '../actions/applications'
import CreateResourceModal from '../components/modals/CreateResourceModal'
import withAccess from '../components/common/withAccess'

const handleCreateResource = (dispatch, yaml) => dispatch(createApplication(RESOURCE_TYPES.HCM_APPLICATIONS, yaml))

const registerApplicationModal = <CreateResourceModal
  key='registerApplication'
  headingTextKey='actions.create.application'
  submitBtnTextKey='actions.create.application'
  onCreateResource={ handleCreateResource }
  resourceDescriptionKey='modal.createresource.application'
/>


export default withRouter(withAccess(typedResourcePageWithListAndDetails(
  RESOURCE_TYPES.HCM_APPLICATIONS,
  ['overview', 'design', 'topology'],
  [registerApplicationModal],
  ['/design', '/topology'],
  [<ResourceTableModule key='placementPolicies' definitionsKey='placementPolicyKeys' />,
    <ResourceTableModule key='deployables' definitionsKey='deployablesKeys' />]
), ROLES.OPERATOR))
