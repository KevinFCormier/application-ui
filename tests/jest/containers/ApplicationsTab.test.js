/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

const React = require("../../../node_modules/react");

import ApplicationsTab from "../../../src-web/containers/ApplicationsTab";

import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import configureMockStore from "redux-mock-store";

import {
  reduxStoreAppPipeline,
  reduxStoreAllAppsPipeline,
  serverProps
} from "../components/TestingData";

const mockStore = configureMockStore();
const storeApp = mockStore(reduxStoreAppPipeline);
const storeAllApps = mockStore(reduxStoreAllAppsPipeline);

const secondaryHeaderProps = {
  title: "routes.applications",
  tabs: [],
  resourceFilters: []
};

const resourceType = {
  name: "QueryApplications",
  list: "QueryApplicationList"
};

describe("ApplicationsTab", () => {
  it("ApplicationsTab renders correctly with data on single app.", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={storeApp}>
            <ApplicationsTab
              serverProps={serverProps}
              secondaryHeaderProps={secondaryHeaderProps}
              resourceType={resourceType}
              status="DONE"
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ApplicationsTab renders correctly with data on all app.", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={storeAllApps}>
            <ApplicationsTab
              serverProps={serverProps}
              secondaryHeaderProps={secondaryHeaderProps}
              resourceType={resourceType}
              status="DONE"
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
