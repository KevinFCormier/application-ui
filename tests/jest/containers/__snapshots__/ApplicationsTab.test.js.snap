// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationsTab ApplicationsTab renders correctly with data on all app. 1`] = `
<div
  className="page-content-container"
  role="main"
>
  <div>
    <div
      id="resource-list"
    >
      <div
        className="bx--data-table-v2-container"
        id="undefined-table-container"
      >
        <div
          className="table-title"
        >
          All applications
           
          <span>
            (
            0
            )
          </span>
        </div>
        <section
          aria-label="undefined-search"
          className="bx--table-toolbar"
          role="region"
        >
          <div
            className="bx--toolbar-search-container"
          >
            <div
              aria-labelledby="undefined-search-label"
              className="bx--search bx--search--sm bx--search--light"
              role="search"
            >
              <svg
                alt="Filter table"
                aria-label="Filter table"
                className="bx--search-magnifier"
                fillRule="evenodd"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
              >
                <title>
                  Filter table
                </title>
                <path
                  d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
                />
              </svg>
              <label
                className="bx--label"
                htmlFor="undefined-search"
                id="undefined-search-label"
              >
                Filter table
              </label>
              <input
                aria-label="undefined-search"
                className="bx--search-input"
                id="undefined-search"
                onChange={[Function]}
                placeholder="Find resources"
                type="text"
                value="aa"
              />
              <button
                className="bx--search-close"
                onClick={[Function]}
                type="button"
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  fillRule="evenodd"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="bx--toolbar-content"
          >
            <div>
              <button
                className="bx--btn bx--btn--sm bx--btn--primary"
                disabled={false}
                id="New application"
                onClick={[Function]}
                tabIndex={0}
                type="button"
              >
                New application
                <svg
                  alt="Provide icon description if icon is used"
                  aria-label="Provide icon description if icon is used"
                  className="bx--btn__icon"
                  fillRule="evenodd"
                  height="16"
                  name="add--glyph"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    Provide icon description if icon is used
                  </title>
                  <path
                    d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </section>
        <table
          className="bx--data-table-v2 resource-table bx--data-table-v2--zebra"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending bx--table-sort-v2--active"
                  data-default-key="name"
                  data-key="name"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Application name
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="namespace"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Namespace
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="clusters"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Managed clusters
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="subscriptions"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Subscriptions (On managed clusters)
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="created"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Created
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <div
        className="bx--pagination"
        disabled={true}
      >
        <div
          className="bx--pagination__left"
        >
          <span
            className="bx--pagination__text"
          >
            items per page
          </span>
          <div
            className="bx--form-item"
          >
            <div
              className="bx--select bx--select--inline"
            >
              <label
                className="bx--label bx--visually-hidden"
                htmlFor="bx-pagination-select-resource-table-pagination"
              >
                items per page
              </label>
              <select
                className="bx--select-input"
                id="bx-pagination-select-resource-table-pagination"
                onChange={[Function]}
                value={20}
              >
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={5}
                >
                  5
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={10}
                >
                  10
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={20}
                >
                  20
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={50}
                >
                  50
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={75}
                >
                  75
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={100}
                >
                  100
                </option>
              </select>
              <svg
                alt="open list of options"
                aria-label="open list of options"
                className="bx--select__arrow"
                fillRule="evenodd"
                height="5"
                role="img"
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  open list of options
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
          <span
            className="bx--pagination__text"
          >
             |  
            {0}-{1} of {0} items
          </span>
        </div>
        <div
          className="bx--pagination__right bx--pagination--inline"
        >
          <span
            className="bx--pagination__text"
          >
            {0} of {1} pages
          </span>
          <button
            aria-label="Backward"
            className="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              alt="Backward"
              aria-label="Backward"
              className="bx--pagination__button-icon"
              fillRule="evenodd"
              height="12"
              role="img"
              viewBox="0 0 7 12"
              width="7"
            >
              <title>
                Backward
              </title>
              <path
                d="M1.45 6.002L7 11.27l-.685.726L0 6.003 6.315 0 7 .726z"
              />
            </svg>
          </button>
          <button
            aria-label="Forward"
            className="bx--pagination__button bx--pagination__button--forward"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              alt="Forward"
              aria-label="Forward"
              className="bx--pagination__button-icon"
              fillRule="evenodd"
              height="12"
              role="img"
              viewBox="0 0 7 12"
              width="7"
            >
              <title>
                Forward
              </title>
              <path
                d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationsTab ApplicationsTab renders correctly with data on single app. 1`] = `
<div
  className="page-content-container"
  role="main"
>
  <div>
    <div
      id="resource-list"
    >
      <div
        className="bx--data-table-v2-container"
        id="undefined-table-container"
      >
        <div
          className="table-title"
        >
          All applications
           
          <span>
            (
            0
            )
          </span>
        </div>
        <section
          aria-label="undefined-search"
          className="bx--table-toolbar"
          role="region"
        >
          <div
            className="bx--toolbar-search-container"
          >
            <div
              aria-labelledby="undefined-search-label"
              className="bx--search bx--search--sm bx--search--light"
              role="search"
            >
              <svg
                alt="Filter table"
                aria-label="Filter table"
                className="bx--search-magnifier"
                fillRule="evenodd"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
              >
                <title>
                  Filter table
                </title>
                <path
                  d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
                />
              </svg>
              <label
                className="bx--label"
                htmlFor="undefined-search"
                id="undefined-search-label"
              >
                Filter table
              </label>
              <input
                aria-label="undefined-search"
                className="bx--search-input"
                id="undefined-search"
                onChange={[Function]}
                placeholder="Find resources"
                type="text"
                value="aa"
              />
              <button
                className="bx--search-close"
                onClick={[Function]}
                type="button"
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  fillRule="evenodd"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="bx--toolbar-content"
          >
            <div>
              <button
                className="bx--btn bx--btn--sm bx--btn--primary"
                disabled={false}
                id="New application"
                onClick={[Function]}
                tabIndex={0}
                type="button"
              >
                New application
                <svg
                  alt="Provide icon description if icon is used"
                  aria-label="Provide icon description if icon is used"
                  className="bx--btn__icon"
                  fillRule="evenodd"
                  height="16"
                  name="add--glyph"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    Provide icon description if icon is used
                  </title>
                  <path
                    d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </section>
        <table
          className="bx--data-table-v2 resource-table bx--data-table-v2--zebra"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending bx--table-sort-v2--active"
                  data-default-key="name"
                  data-key="name"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Application name
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="namespace"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Namespace
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="clusters"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Managed clusters
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="subscriptions"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Subscriptions (On managed clusters)
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              >
                <button
                  className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                  data-default-key="name"
                  data-key="created"
                  onClick={[Function]}
                  title="Sort rows by this header in descending order"
                >
                  <span
                    className="bx--table-header-label"
                  >
                    Created
                  </span>
                  <svg
                    alt="Sort rows by this header in descending order"
                    aria-label="Sort rows by this header in descending order"
                    className="bx--table-sort-v2__icon"
                    fillRule="evenodd"
                    height="5"
                    name="caret--down"
                    role="img"
                    viewBox="0 0 10 5"
                    width="10"
                  >
                    <title>
                      Sort rows by this header in descending order
                    </title>
                    <path
                      d="M0 0l5 4.998L10 0z"
                    />
                  </svg>
                </button>
              </th>
              <th
                scope="col"
              />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <div
        className="bx--pagination"
        disabled={true}
      >
        <div
          className="bx--pagination__left"
        >
          <span
            className="bx--pagination__text"
          >
            items per page
          </span>
          <div
            className="bx--form-item"
          >
            <div
              className="bx--select bx--select--inline"
            >
              <label
                className="bx--label bx--visually-hidden"
                htmlFor="bx-pagination-select-resource-table-pagination"
              >
                items per page
              </label>
              <select
                className="bx--select-input"
                id="bx-pagination-select-resource-table-pagination"
                onChange={[Function]}
                value={20}
              >
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={5}
                >
                  5
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={10}
                >
                  10
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={20}
                >
                  20
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={50}
                >
                  50
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={75}
                >
                  75
                </option>
                <option
                  className="bx--select-option"
                  disabled={false}
                  hidden={false}
                  value={100}
                >
                  100
                </option>
              </select>
              <svg
                alt="open list of options"
                aria-label="open list of options"
                className="bx--select__arrow"
                fillRule="evenodd"
                height="5"
                role="img"
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  open list of options
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
          <span
            className="bx--pagination__text"
          >
             |  
            {0}-{1} of {0} items
          </span>
        </div>
        <div
          className="bx--pagination__right bx--pagination--inline"
        >
          <span
            className="bx--pagination__text"
          >
            {0} of {1} pages
          </span>
          <button
            aria-label="Backward"
            className="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              alt="Backward"
              aria-label="Backward"
              className="bx--pagination__button-icon"
              fillRule="evenodd"
              height="12"
              role="img"
              viewBox="0 0 7 12"
              width="7"
            >
              <title>
                Backward
              </title>
              <path
                d="M1.45 6.002L7 11.27l-.685.726L0 6.003 6.315 0 7 .726z"
              />
            </svg>
          </button>
          <button
            aria-label="Forward"
            className="bx--pagination__button bx--pagination__button--forward"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              alt="Forward"
              aria-label="Forward"
              className="bx--pagination__button-icon"
              fillRule="evenodd"
              height="12"
              role="img"
              viewBox="0 0 7 12"
              width="7"
            >
              <title>
                Forward
              </title>
              <path
                d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
